src/toil/test/cwl/cwlTest.py

```diff
@@ -414,46 +401,27 @@ def tearDown(self):
             shutil.rmtree(self.outDir)
         unittest.TestCase.tearDown(self)
 
+    def refresh_deleted_fixtures(self):
+        # TODO: Prevent Toil from deleting these?
+        for fixture in ['tests/testdir/a', 'tests/testdir/b', 'tests/testdir/c/d']:
+            if not os.path.exists(os.path.join(self.cwlSpec, os.path.dirname(fixture))):
+                os.makedirs(os.path.join(self.cwlSpec, os.path.dirname(fixture)), exist_ok=True)
+                with open(os.path.join(self.cwlSpec, os.path.dirname(fixture)), 'w') as f:
+                    f.write()
+
     @slow
     @pytest.mark.timeout(CONFORMANCE_TEST_TIMEOUT)
-    # Cannot work until we fix https://github.com/DataBiosphere/toil/issues/2801
-    @pytest.mark.xfail
     def test_run_conformance_with_caching(self):
         self.test_run_conformance(caching=True)
 
     @slow
     @pytest.mark.timeout(CONFORMANCE_TEST_TIMEOUT)
     def test_run_conformance(self, batchSystem=None, caching=False):
-        try:
-            # TODO: we do not currently pass tests: 237 (offset from other versions), 307, 309, 310, 311, 330, 331, 332
-            selected_tests = '1-236,238-306,308,312-329,333-336'
-            cmd = [f'cwltest',
-                   f'--tool=toil-cwl-runner',
-                   f'--test={self.test_yaml}',
-                   f'--timeout=2400',
-                   f'--basedir={self.cwlSpec}',
-                   f'-n={selected_tests}']
-            if batchSystem:
-                cmd.extend(["--batchSystem", batchSystem])
-
-            args_passed_directly_to_toil = ['--enable-dev']
-            args_passed_directly_to_toil.extend(['--disableCaching={}'.format(not caching)])
-            cmd.extend(['--'] + args_passed_directly_to_toil)
-            
-            logger.info("Running: '%s'", "' '".join(cmd))
-            subprocess.check_output(cmd, cwd=self.cwlSpec, stderr=subprocess.STDOUT)
-        except subprocess.CalledProcessError as e:
-            only_unsupported = False
-            # check output -- if we failed but only have unsupported features, we're okay
-            p = re.compile(r"(?P<failures>\d+) failures, (?P<unsupported>\d+) unsupported features")
-
-            error_log = e.output.decode('utf-8')
-            for line in error_log.split('\n'):
-                m = p.search(line)
-                if m:
-                    if int(m.group("failures")) == 0 and int(m.group("unsupported")) > 0:
-                        only_unsupported = True
-                        break
-            if not only_unsupported:
-                print(error_log)
-                raise e
+        # TODO: we do not currently pass tests: 237 (offset from other versions), 307, 309, 310, 311, 332
+        for selected_tests in ['1-236,238-306,308,312-329,333-336', '330', '331']:
```

Comment:
Is running tests 330 and 331 last needed? If so, the comment should document why.

Reply:
Yes, each time the fixtures have to be recreated.  Will add a comment!

Reply:
Added.